Library/Homebrew/os.rb

```diff
@@ -49,8 +49,8 @@ def self.kernel_name
   LINUX_CI_OS_VERSION = "Ubuntu 22.04"
   LINUX_GLIBC_CI_VERSION = "2.35"
   LINUX_GLIBC_NEXT_CI_VERSION = "2.39"
-  LINUX_GCC_CI_VERSION = "11.0"
-  LINUX_PREFERRED_GCC_COMPILER_FORMULA = "gcc@11" # https://packages.ubuntu.com/jammy/gcc
+  LINUX_GCC_CI_VERSION = "12.0"
+  LINUX_PREFERRED_GCC_COMPILER_FORMULA = "gcc@12" # https://packages.ubuntu.com/jammy/gcc-12
```

Comment:
We probably need to switch this to check libstdc+++ works so that Ubuntu 22.04 base install still works and only require a compiler when building from source.

The base Ubuntu 22.04 install has libstdc++ 12 by default (just with GCC 11 as a compiler).

Reply:
Is the GCC version available in binary? Or do we want to use GLIBCXX version (3.4.30) or soversion (6.0.30)?

Reply:
Maybe there's a mention in `.comment` but I would not rely on it. GLIBCXX and soversion do map to a GCC version as you can see here: https://gcc.gnu.org/onlinedocs/libstdc++/manual/abi.html (*.30 is GCC 12.1+).

libgcc also has symbol versions but there was nothing new in GCC 12 anyway (https://github.com/gcc-mirror/gcc/blob/master/libgcc/libgcc-std.ver.in)